src/components/autoSuggest/index.test.js

```diff
@@ -0,0 +1,138 @@
+import React from 'react';
+import { expect } from 'chai';
+import { mount } from 'enzyme';
+import { I18nextProvider } from 'react-i18next';
+import { spy } from 'sinon';
+import AutoSuggest from './index';
+import i18n from '../../i18n';
+
+import * as searchActions from './../search/keyAction';
+
+import routes from '../../constants/routes';
+import keyCodes from './../../constants/keyCodes';
+import mockSearchResults from './searchResults.mock';
+
+describe('AutoSuggest', () => {
+  let wrapper;
+  let props;
+  let results;
+  let submitSearchSpy;
+  let visitAndSaveSearchSpy;
+
+  beforeEach(() => {
+    results = { ...mockSearchResults };
+    props = {
+      t: key => key,
+      history: {
+        push: spy(),
+      },
+      results,
+    };
+
+    submitSearchSpy = spy(AutoSuggest.prototype, 'submitSearch');
+    visitAndSaveSearchSpy = spy(searchActions, 'visitAndSaveSearch');
+    wrapper = mount(<I18nextProvider i18n={i18n}><AutoSuggest {...props} /></I18nextProvider>);
+    wrapper.update();
+  });
+
+  afterEach(() => {
+    submitSearchSpy.restore();
+    visitAndSaveSearchSpy.restore();
+  });
+
+  it('should render a row for each entity found {addresses,delegates,transactions}', () => {
+    expect(wrapper).to.have.exactly(3).descendants('.addresses-result');
+    expect(wrapper).to.have.exactly(3).descendants('.delegates-result');
+    expect(wrapper).to.have.exactly(3).descendants('.transactions-result');
+  });
+
+  it('should not render any row for not found entities {delegates}', () => {
+    const partialResults = {
+      ...results,
+      addresses: [],
+      transactions: [],
+    };
+    const updatedProps = { ...props, results: partialResults };
+    wrapper = mount(<I18nextProvider i18n={i18n}>
+      <AutoSuggest {...updatedProps} /></I18nextProvider>);
+    wrapper.update();
+    expect(wrapper).to.have.exactly(0).descendants('.addresses-result');
+    expect(wrapper).to.have.exactly(3).descendants('.delegates-result');
+    expect(wrapper).to.have.exactly(0).descendants('.transactions-result');
+  });
+
+  it('should show autosuggest on search input change and hide it on blur', () => {
+    let autosuggestDropdown = wrapper.find('.autosuggest-dropdown').first();
+    const autosuggestInput = wrapper.find('.autosuggest-input').find('input').first();
+    expect(autosuggestDropdown.props().className.match(new RegExp(/ {1}autoSuggest__show__/g))).to.be.equal(null);
+    autosuggestInput.simulate('focus');
+    autosuggestInput.simulate('change', { target: { value: 'abc' } });
+    wrapper.update();
+    autosuggestDropdown = wrapper.find('.autosuggest-dropdown').first();
+    expect(autosuggestDropdown.props().className.match(new RegExp(/ {1}autoSuggest__show__/g))).not.to.be.equal(null);
```

Comment:
It will give better fail messages than `null expected to be not null` if changed to
`expect(autosuggestDropdown.props().className).to.match(new RegExp(/ {1}autoSuggest__show__/g));`

or even better with https://github.com/producthunt/chai-enzyme#classnamestr :
` import styles from './autoSuggest.css'`
`expect(autosuggestDropdown).to.have.className(styles.show);`